exercises/pythagorean-triplet/test/test_pythagorean_triplet.c

```diff
@@ -0,0 +1,158 @@
+#include "vendor/unity.h"
+#include "../src/pythagorean_triplet.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+static void check_count(size_t expected, size_t actual)
+{
+   TEST_ASSERT_EQUAL_UINT64(expected, actual);
+}
+
+static void check_triplets(size_t count, uint16_t expected[][3],
+                           uint16_t actual[][3])
+{
+   for (size_t i = 0; i < count; ++i) {
+      bool triplet_found = false;
+      for (size_t j = 0; j < count; ++j) {
+         for (size_t k = 0; k < LENGTH_OF_TRIPLET; ++k) {
+            if (expected[i][k] != actual[j][k])
+               triplet_found = false;
+            else
+               triplet_found = true;
+         }
+         if (triplet_found)
+            break;
+      }
+      if (!triplet_found)
+         TEST_FAIL();
+   }
+}
```

Comment:
I think this should have a check for the number of triplets in `actual`. Without checking the count of the provided triplets you can't guarantee that there aren't extraneous results and you can't ensure that the test code will not read out of bounds.

Reply:
The count check is separate (the method immediately above this in the file) as some tests require _just_ a count check. AFAIK there's no way to check that an array actually has the bounds indicated by a count variable though?

Reply:
You're absolutely right. I must have glossed over that. Sorry!